'HubAdapter' is deprecated & 'LoadClass' is deprecated &  unsafe use of a nullable receiver of type Thread.UncaughtExceptionHandler

### Platform

Flutter Mobile Android

### Obfuscation

Disabled

### Debug Info

Disabled

### Doctor

flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.35.1, on Microsoft Windows [Version 10.0.26100.4946], locale en-US)
[√] Windows Version (Windows 11 or higher, 24H2, 2009)
[√] Android toolchain - develop for Android devices (Android SDK version 36.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.13.6)
[√] Android Studio (version 2025.1.2)
[√] VS Code (version 1.103.2)
[√] Connected device (3 available)
[√] Network resources

• No issues found!

### Version

9.6.0

### Steps to Reproduce

ran "flutter build apk --release" in windows 11 and while building release apk i got these exception:

exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutter.kt:75:50: warning: 'toUpperCase(Locale): String' is deprecated. Use uppercase() instead.
exception:         val sentryLevel = SentryLevel.valueOf(it.toUpperCase(Locale.ROOT))
exception:                                                  ^^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutter.kt:168:37: warning: 'toUpperCase(Locale): String' is deprecated. Use uppercase() instead.
exception:                     Type.valueOf(it.toUpperCase(Locale.ROOT))
exception:                                     ^^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:18:18: warning: 'HubAdapter' is deprecated. Deprecated in Java
exception: import io.sentry.HubAdapter
exception:                  ^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:21:31: warning: 'LoadClass' is deprecated. Deprecated in Java
exception: import io.sentry.android.core.LoadClass
exception:                               ^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:235:13: warning: 'getter for defaultDisplay: Display!' is deprecated. Deprecated in Java
exception:           ?.defaultDisplay
exception:             ^^^^^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:291:21: warning: 'HubAdapter' is deprecated. Deprecated in Java
exception:       val options = HubAdapter.getInstance().options
exception:                     ^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:305:21: warning: 'HubAdapter' is deprecated. Deprecated in Java
exception:       val options = HubAdapter.getInstance().options
exception:                     ^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:401:19: warning: 'HubAdapter' is deprecated. Deprecated in Java
exception:     val options = HubAdapter.getInstance().options as SentryAndroidOptions
exception:                   ^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:435:5: warning: 'HubAdapter' is deprecated. Deprecated in Java
exception:     HubAdapter.getInstance().close()
exception:     ^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:452:7: warning: unsafe use of a nullable receiver of type Thread.UncaughtExceptionHandler?
exception:       mainThread.uncaughtExceptionHandler.uncaughtException(mainThread, exception)
exception:       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
exception: c:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dev\sentry_flutter-9.6.0\android\src\main\kotlin\io\sentry\flutter\SentryFlutterPlugin.kt:467:19: warning: 'HubAdapter' is deprecated. Deprecated in Java
exception:     val options = HubAdapter.getInstance().options
exception:                   ^^^^^^^^^^


### Expected Result

build without exception and error in terminal for fist time.

### Actual Result

at first i built the apk and got the exeptions, after got exeption, retry and didn't get any exepetion or warning in terminal

### Are you willing to submit a PR?

None

--- a/packages/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutter.kt
+++ b/packages/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutter.kt
@@ -72,7 +72,7 @@ class SentryFlutter {
     }
     data.getIfNotNull<String>("diagnosticLevel") {
       if (options.isDebug) {
-        val sentryLevel = SentryLevel.valueOf(it.toUpperCase(Locale.ROOT))
+        val sentryLevel = SentryLevel.valueOf(it.uppercase())
         options.setDiagnosticLevel(sentryLevel)
       }
     }
@@ -165,7 +165,7 @@ class SentryFlutter {
               ?.let {
                 type =
                   try {
-                    Type.valueOf(it.toUpperCase(Locale.ROOT))
+                    Type.valueOf(it.uppercase())
                   } catch (_: IllegalArgumentException) {
                     Log.w("Sentry", "Could not parse `type` from proxy json: $proxyJson")
                     null

--- a/packages/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt
+++ b/packages/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt
@@ -15,10 +15,9 @@ import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 import io.sentry.Breadcrumb
 import io.sentry.DateUtils
-import io.sentry.HubAdapter
+import io.sentry.ScopesAdapter
 import io.sentry.Sentry
 import io.sentry.android.core.InternalSentrySdk
-import io.sentry.android.core.LoadClass
 import io.sentry.android.core.SentryAndroid
 import io.sentry.android.core.SentryAndroidOptions
 import io.sentry.android.core.performance.AppStartMetrics
@@ -26,7 +25,6 @@ import io.sentry.android.core.performance.TimeSpan
 import io.sentry.android.replay.ReplayIntegration
 import io.sentry.android.replay.ScreenshotRecorderConfig
 import io.sentry.protocol.DebugImage
-import io.sentry.protocol.SentryId
 import io.sentry.protocol.User
 import io.sentry.transport.CurrentDateProvider
 import org.json.JSONObject
@@ -289,7 +287,7 @@ class SentryFlutterPlugin :
     result: Result,
   ) {
     if (user != null) {
-      val options = HubAdapter.getInstance().options
+      val options = ScopesAdapter.getInstance().options
       val userInstance = User.fromMap(user, options)
       Sentry.setUser(userInstance)
     } else {
@@ -303,7 +301,7 @@ class SentryFlutterPlugin :
     result: Result,
   ) {
     if (breadcrumb != null) {
-      val options = HubAdapter.getInstance().options
+      val options = ScopesAdapter.getInstance().options
       val breadcrumbInstance = Breadcrumb.fromMap(breadcrumb, options)
       Sentry.addBreadcrumb(breadcrumbInstance)
     }
@@ -371,7 +369,7 @@ class SentryFlutterPlugin :
   }
 
   private fun closeNativeSdk(result: Result) {
-    HubAdapter.getInstance().close()
+    ScopesAdapter.getInstance().close()
 
     result.success("")
   }
@@ -385,15 +383,17 @@ class SentryFlutterPlugin :
 
     private const val NATIVE_CRASH_WAIT_TIME = 500L
 
+    @Suppress("unused") // Used by native/jni bindings
     @JvmStatic
     fun privateSentryGetReplayIntegration(): ReplayIntegration? = replay
 
     @JvmStatic
     fun getApplicationContext(): Context? = applicationContext
 
+    @Suppress("unused") // Used by native/jni bindings
     @JvmStatic
     fun loadContextsAsBytes(): ByteArray? {
-      val options = HubAdapter.getInstance().options
+      val options = ScopesAdapter.getInstance().options
       val context = getApplicationContext()
       if (options !is SentryAndroidOptions || context == null) {
         return null
@@ -409,9 +409,10 @@ class SentryFlutterPlugin :
       return json.toByteArray(Charsets.UTF_8)
     }
 
+    @Suppress("unused") // Used by native/jni bindings
     @JvmStatic
     fun loadDebugImagesAsBytes(addresses: Set<String>): ByteArray? {
-      val options = HubAdapter.getInstance().options as SentryAndroidOptions
+      val options = ScopesAdapter.getInstance().options as SentryAndroidOptions
 
       val debugImages =
         if (addresses.isEmpty()) {
@@ -447,7 +448,7 @@ class SentryFlutterPlugin :
     private fun crash() {
       val exception = RuntimeException("FlutterSentry Native Integration: Sample RuntimeException")
       val mainThread = Looper.getMainLooper().thread
-      mainThread.uncaughtExceptionHandler.uncaughtException(mainThread, exception)
+      mainThread.uncaughtExceptionHandler?.uncaughtException(mainThread, exception)
       mainThread.join(NATIVE_CRASH_WAIT_TIME)
     }
 

--- a/packages/flutter/example/android/app/build.gradle
+++ b/packages/flutter/example/android/app/build.gradle
@@ -34,7 +34,7 @@ android {
     }
     kotlinOptions {
         jvmTarget = JavaVersion.VERSION_1_8
-        languageVersion = "1.4"
+        languageVersion = "1.8"
     }
 
     compileSdkVersion 35

